### PR TITLE
Improve calendar styling

### LINKED ISSUE #1391: Custom Calendar Styling
## Feature Description

This is a followup to #789 which is a short term solution for the Beta Release. 
Once a calendar is implemented we will need to style it according to the [designs](https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=6113%3A618)

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria
1. The calendar should match the [designs in Figma ](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3408%3A85035)

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

It has always bugged me that our date picker's styling does not match the rest of the editor.

## Summary

Updates the date/time picker in the editor to better match the rest of the app.

## To-do

- [ ] Fix tests

## User-facing changes

**Before**

Screenshot 2022-10-17 at 09 31 51
Screenshot 2022-10-17 at 09 32 07
Screenshot 2022-10-17 at 09 32 00

**After**

Screenshot 2022-10-17 at 10 26 02
Screenshot 2022-10-17 at 10 25 56
Screenshot 2022-10-17 at 10 25 51

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Date picker should still work the same, including the time input & the AM/PM toggle

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #1391

### CODE REVIEW

File: packages/story-editor/src/components/form/dateTime/timeZone.js [hunk 1]
```diff
@@ -65,7 +65,13 @@ function TimeZone() {
 
   return (
     <Wrapper>
-      <Tooltip hasTail title={tooltip} placement={PLACEMENT.TOP}>
+      <Tooltip
+        hasTail
+        title={tooltip}
+        placement={PLACEMENT.TOP}
+        // Higher than popup z-index in PublishTime.
+        popupZIndexOverride={11}
```
Comment (miina) [inline packages/story-editor/src/components/form/dateTime/timeZone.js]: Maybe add to `constants/zIndex.js` instead.

**Review by miina — COMMENTED** [review]
Screenshot 2022-10-17 at 17 26 51

^ The contrast between the background color and the hovered / focused date seems a bit low -- perhaps it has been tested and is ok though.

**Comment by swissspidy** [comment]
> Screenshot 2022-10-17 at 17 26 51
> 
> ^ The contrast between the background color and the hovered / focused date seems a bit low -- perhaps it has been tested and is ok though.

Good call. Improved the contrast now.

**Review by miina — APPROVED** [review]

**Comment by kkalarickal** [comment]
@swissspidy - there seems to be a different color for current date vs the selected date which is lost with the changes in this ticket.

For example, consider a story with publish data 01-Nov-2022. This is different from today's date of 17-Oct-2022

In the `main branch` (without these changes), the date picker show a yellow shade on 17-Oct-2022 and a blue shade on 01-Nov-2022


However, with the changes in `12481` applied, the two colors are the same:

Similarly, when you go one level higher by clicking the Month-Year title, you can see that in `main branch` the current month is yellow while the selected month is blue.

But in `12481 branch` the color for selected month is missing !!

Since this is a ticket for consistent calendar styling, these color changes for present vs selected dates should probably be fixed.

**Comment by swissspidy** [comment]
> However, with the changes in 12481 applied, the two colors are the same:

Not technically :-) Well, on hover they were _similar_, but otherwise the current date has a gray bg:

Screenshot 2022-10-18 at 08 40 39

> But in 12481 branch the color for selected month is missing

Good catch 👍 

Fixed now:

Screenshot 2022-10-18 at 08 55 48

**Comment by kkalarickal** [comment]
Confirmed that the current date/month/year appears in a grey box when the published date/month/year is different

**Pull request merged**